src/riak_kv_ts_api.erl

```diff
@@ -101,8 +101,12 @@ api_calls() ->
                    {ok, riak_kv_qry:query_tabular_result()} |
                    {error, term()}.
 query(QueryStringOrSQL, DDL) ->
-    riak_kv_qry:submit(QueryStringOrSQL, DDL).
-
+    case riak_kv_qry:submit(QueryStringOrSQL, DDL) of
+        {error, {impossible_where_clause, _}} ->
+            {ok, {[],[], []}};
```

Comment:
We now have three types of where clause:
1. A valid where clause which covers the partition key
2. An invalid clause that does not cover the partition key
3. A valid clause that covers the partition key but we can pre-determine that it won't return results.

2 and 3 are quite different conditions. 2 is valid SQL which would return results in a relational DB but Riak TS cannot because of its requirements to cover the partition key, we have to return an error or an incorrect result because we can't access the actual data.

3 is valid, and in TS 1.5 it would be run but we can pre-determine that it will never return any results, so as an optimisation, we return the known result without hitting the vnodes.